src/__tests__/global-container.test.ts

```diff
@@ -782,3 +783,60 @@ test("predicateAwareClassFactory returns new instances each call with caching of
 
   expect(factory(globalContainer)).not.toBe(factory(globalContainer));
 });
+
+describe("dispose", () => {
+  class Foo implements Disposable {
+    disposed = false;
+    dispose(): void {
+      this.disposed = true;
+    }
+  }
+  class Bar implements Disposable {
+    disposed = false;
+    dispose(): void {
+      this.disposed = true;
+    }
+  }
+
+  it("renders the container useless", () => {
+    const container = globalContainer.createChildContainer();
+    container.dispose();
+
+    expect(() => container.reset()).toThrow(/disposed/);
+  });
+
+  it("disposes all child disposables", () => {
+    const container = globalContainer.createChildContainer();
+
+    const foo = container.resolve(Foo);
+    const bar = container.resolve(Bar);
+
+    container.dispose();
+
+    expect(foo.disposed).toBeTruthy();
+    expect(bar.disposed).toBeTruthy();
+  });
+
+  it("disposes all instances of the same type", () => {
```

Comment:
Question - what happens for array-type (`@injectAll` type) resolutions - do we resolve each element in the array?